app/stac_api/views.py

```diff
@@ -363,10 +364,13 @@ def get_queryset(self):
         )
 
     def get(self, request, *args, **kwargs):
-        queryset = self.filter_queryset(self.get_queryset())
-        if not Collection.objects.filter(name=self.kwargs['collection_name']).exists() or \
-                not Item.objects.filter(name=self.kwargs['item_name']).exists():
+        if (not Collection.objects.filter(name=self.kwargs['collection_name']).exists() or
+                not Item.objects.filter(name=self.kwargs['item_name']).exists()):
+            logger.error("Either the collection (%s) or the item (%s) do not exist",
+                         self.kwargs['collection_name'], self.kwargs['item_name'])
             raise Http404("This collection does not exists.")
```

Comment:
Formatting issue and the message of the Http404 is not always correct, I mean it is either the collection and/or the item that doesn't exist. Actually here it would be better to have two separate if with each one the correct error message. Don't forget these error messages are sent to customer with the 404.